tests/genai/judges/test_judge_tool_registry.py

```diff
@@ -0,0 +1,192 @@
+        request_time=1234567890,
+        state=TraceState.OK,
+        execution_duration=100,
+    )
+    trace = Trace(info=trace_info, data=None)
+
+    tool_call = ToolCall(
+        function=FunctionToolCallArguments(
+            name="mock_tool", arguments=json.dumps({"param": "value"})
+        )
+    )
+
+    result = registry.invoke(tool_call, trace)
+    assert result == "mock_result_with_1_args"
+
+
+def test_registry_invoke_tool_not_found():
+    registry = JudgeToolRegistry()
+
+    trace_info = TraceInfo(
+        trace_id="test-trace-id",
+        trace_location=TraceLocation.from_experiment_id("0"),
+        request_time=1234567890,
+        state=TraceState.OK,
+        execution_duration=100,
+    )
+    trace = Trace(info=trace_info, data=None)
+
+    tool_call = ToolCall(
+        function=FunctionToolCallArguments(name="nonexistent_tool", arguments=json.dumps({}))
+    )
+
+    with pytest.raises(MlflowException, match="Tool 'nonexistent_tool' not found in registry"):
+        registry.invoke(tool_call, trace)
+
+
+def test_registry_invoke_tool_invalid_json():
+    registry = JudgeToolRegistry()
+    mock_tool = MockTool()
+    registry.register(mock_tool)
+
+    trace_info = TraceInfo(
+        trace_id="test-trace-id",
+        trace_location=TraceLocation.from_experiment_id("0"),
+        request_time=1234567890,
+        state=TraceState.OK,
+        execution_duration=100,
+    )
+    trace = Trace(info=trace_info, data=None)
+
+    tool_call = ToolCall(
+        function=FunctionToolCallArguments(name="mock_tool", arguments="invalid json {{")
+    )
+
+    with pytest.raises(MlflowException, match="Invalid JSON arguments for tool 'mock_tool'"):
+        registry.invoke(tool_call, trace)
+
+
+def test_registry_invoke_tool_invalid_arguments():
+    registry = JudgeToolRegistry()
+
+    class StrictTool(JudgeTool):
+        @property
+        def name(self) -> str:
+            return "strict_tool"
+
+        def get_definition(self) -> ToolDefinition:
+            return ToolDefinition(function={}, type="function")
+
+        def invoke(self, trace: Trace, required_param: str) -> str:
+            return f"result_{required_param}"
+
+    strict_tool = StrictTool()
+    registry.register(strict_tool)
+
+    trace_info = TraceInfo(
+        trace_id="test-trace-id",
+        trace_location=TraceLocation.from_experiment_id("0"),
+        request_time=1234567890,
+        state=TraceState.OK,
+        execution_duration=100,
+    )
+    trace = Trace(info=trace_info, data=None)
+
+    tool_call = ToolCall(
+        function=FunctionToolCallArguments(name="strict_tool", arguments=json.dumps({}))
+    )
+
+    with pytest.raises(MlflowException, match="Invalid arguments for tool 'strict_tool'"):
+        registry.invoke(tool_call, trace)
+
+
+def test_global_functions_work():
+    from mlflow.genai.judges.tools.registry import _judge_tool_registry
+
+    original_tools = _judge_tool_registry.list_tools()
```

Comment:
Ensuring that it's empty or that it isn't mutated at the end of the test?

Reply:
Done! :D